packages/kit/src/runtime/server/page/fetch.js

```diff
@@ -63,7 +63,7 @@ export function create_fetch({ event, options, state, route, prerender_default }
 			}
 		}
 
-		const resolved = resolve(event.url.pathname, requested.split('?')[0]);
+		const resolved = resolve(event.url.pathname, requested.split('?')[0]).replace(/#.+$/, '');
```

Comment:
would this theoretically result in a URL which exists? Also, this probably doesn't result in the same error as in the browser (404 vs some error about invalid urls). Are we ok with this difference in behavior?

Reply:
Not sure I follow? You can totally do `fetch('/foo#bar')` from the browser, the `#bar` just gets ignored. This emulates that behaviour

Reply:
I read the related issue as "the browser fetch says this URL is invalid, SvelteKit's fetch can deal with it fine". This change just removes everything after (and including) the `#`, which is different from throwing an "invalid url" error. It could also mean that the fetch ends up fetching something that actually exists.

Reply:
A URL with a `#` in it is totally valid with `fetch`, it just gets ignored. The confusing error in that issue was because the OP was calling `fetch` in the `<script>` block, not in `onMount`, and undici/node-fetch chokes because it's a root-relative URL

Reply:
Ah ok, all good then